### PR TITLE
Implement Spread Properties

### PR DESCRIPTION
**Pull request by arv**
https://github.com/sebmarkbage/ecmascript-rest-spread/blob/master/Spec.md#spread-properties

This add support for ... in object literals which is currently at stage 2.

```js
var x = {a: 1};
var y = {...a, b: 2};  // {a: 1, b: 2}
```

### CODE REVIEW

**Comment by arv** [comment]
@johnjbarton PTAL

**Comment by arv** [comment]
@johnjbarton ping?

**Comment by johnjbarton** [comment]
Sorry, looks like my email forwarding is busted.

File: src/codegeneration/SpreadPropertiesTransformer.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import {createObjectLiteral, createArgumentList} from './ParseTreeFactory.js';
+import {parseExpression} from './PlaceholderParser.js';
+import {ParseTreeTransformer} from './ParseTreeTransformer.js';
+
+function hasSpread(trees) {
+  return trees.some((tree) => tree && tree.type === SPREAD_EXPRESSION);
+}
+
+/**
+ * Adds suppor for spread properties on object literals
```
Comment (johnjbarton) [inline src/codegeneration/SpreadPropertiesTransformer.js]: support

File: src/runtime/modules/spreadProperties.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    value: v,
+    writable: true,
+  });
+}
+
+function copyDataProperties(target, source) {
+  if (source == null) {  // needs to be ==
+    return;
+  }
+  // toObject is called by getOwnProperty*
```
Comment (johnjbarton) [inline src/runtime/modules/spreadProperties.js]: ?

Comment (johnjbarton) [inline src/runtime/modules/spreadProperties.js]: Maybe this comment is related to the conversion operation, but the two are a bit cryptic this way.

**Comment by johnjbarton** [comment]
LGTM  I guess this also improves jsx usability?

File: src/runtime/modules/spreadProperties.js (hunk 2 above)
Comment (arv) [inline src/runtime/modules/spreadProperties.js]: Clarified

**Pull request merged**